src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -123,6 +145,55 @@ internal static class GoToDefinitionHelpers
             }
         }
 
+        public static async Task<IEnumerable<INavigableItem>> FindExternalDefinitionsAsync(Document document, int position, IEnumerable<Lazy<INavigableDefinitionProvider>> externalDefinitionProviders, CancellationToken cancellationToken)
+        {
+            foreach (var definitionProvider in externalDefinitionProviders)
+            {
+                var definitions = await definitionProvider.Value.FindDefinitionsAsync(document, position, cancellationToken).ConfigureAwait(false);
+                if (definitions != null && definitions.Any())
+                {
+                    var preferredDefinitions = NavigableItemFactory.GetPreferredNavigableItems(document.Project.Solution, definitions);
+                    if (preferredDefinitions.Any())
+                    {
+                        return preferredDefinitions;
+                    }
+                }
+            }
+
+            return SpecializedCollections.EmptyEnumerable<INavigableItem>();
+        }
+
+        public static async Task<IEnumerable<INavigableItem>> FindExternalDefinitionsAsync(ISymbol symbol, Project project, IEnumerable<Lazy<INavigableDefinitionProvider>> externalDefinitionProviders, CancellationToken cancellationToken)
+        {
+            foreach (var definitionProvider in externalDefinitionProviders)
+            {
+                var definitions = await definitionProvider.Value.FindDefinitionsAsync(symbol, project, cancellationToken).ConfigureAwait(false);
+                if (definitions != null && definitions.Any())
+                {
+                    var preferredDefinitions = NavigableItemFactory.GetPreferredNavigableItems(project.Solution, definitions);
+                    if (preferredDefinitions.Any())
```

Comment:
Same here, we want to return results from the first provider returning at least one navigable item.